docs/docs/connect/data-source/gcs.md

```diff
@@ -36,27 +39,71 @@ If you are using a service account, you will need to run the following command:
 gcloud auth activate-service-account --key-file=path_to_json_key_file
 ```
 :::
-
 You have now configured Google Cloud access from your local environment. Rill will detect and use your credentials the next time you try to ingest a source.
 
-:::tip Did you know?
-If this project has already been deployed to Rill Cloud and credentials have been set for this source, you can use `rill env pull` to [pull these cloud credentials](/connect/credentials/#rill-env-pull) locally (into your local `.env` file). Please note that this may override any credentials you have set locally for this source.
+### Service Account JSON 
+
+Alternatively, we recommend creating explicit credentials in plug into Rill as this will facilitate an easier deploy to Rill Cloud. Assuming you've followed step 1 and 2 above, you'll need to create your json credential with the following command.
+
+```bash
+gcloud iam service-accounts keys create ~/key.json \
+  --iam-account=my-service-account@PROJECT_ID.iam.gserviceaccount.com
+```
+
+:::note Permission denied?
+You'll need to contact your internal cloud admin to create your JSON credentials for you.
 :::
 
-## Separating Dev and Prod Environments
+To configure Rill to use these credentials, create a `.env` file in your project directory (if one doesn't already exist) and add your service account JSON credentials as a single-line string:
 
-When ingesting data locally, consider setting parameters in your connector file to limit how much data is retrieved, since costs can scale with the data source. This also helps other developers clone the project and iterate quickly by reducing ingestion time.
+```bash
+google_application_credentials='{"type": "service_account", "project_id": "your-project", ...}'
+```
 
-For more details, see our [Dev/Prod setup docs](/connect/templating).
+Once configured, Rill will automatically use these credentials for all Google Cloud Platform connections, including [BigQuery](/connect/data-source/bigquery).
 
-## Rill Cloud Deployment
+:::tip Cloud Credentials Management
+If your project has already been deployed to Rill Cloud with configured credentials, you can use `rill env pull` to [retrieve and sync these cloud credentials](/connect/credentials/#rill-env-pull) to your local `.env` file. Note that this operation will overwrite any existing local credentials for this source.
+:::
+
+### HMAC Keys
```

Comment:
I notice we don't have HMAC documentation on the BigQuery page. Should we? Or should we have one central page to discuss authentication to GCP?